test/e2e/common_suite_test.go

```diff
@@ -249,16 +275,93 @@ func (tc *testCase) run() {
 
 func newTestCase(t *testing.T, testName string, assert CloudAssert, assessMessage string) *testCase {
 	testCase := &testCase{
-		testing:       t,
-		testName:      testName,
-		assert:        assert,
-		assessMessage: assessMessage,
-		podState:      v1.PodRunning,
+		testing:        t,
+		testName:       testName,
+		assert:         assert,
+		assessMessage:  assessMessage,
+		podState:       v1.PodRunning,
+		imagePullTimer: false,
 	}
 
 	return testCase
 }
 
+func reverseSlice(slice []string) []string {
+	length := len(slice)
+	for i := 0; i < length/2; i++ {
+		slice[i], slice[length-i-1] = slice[length-i-1], slice[i]
+	}
+	return slice
+}
+
+// timeExtractor for comparing and extracting time from a Log String
+func timeExtractor(log string) (string, error) {
+	matchString := regexp.MustCompile(`\b(\d{2}):(\d{2}):(\d{2})\b`).FindStringSubmatch(log)
+	if len(matchString) != 4 {
+		return "", errors.New("Invalid Time Data")
+	}
+	return matchString[0], nil
+}
+
+func watchImagePullTime(ctx context.Context, client klient.Client, caaPod v1.Pod, Pod v1.Pod) (string, error) {
+	pullingtime := ""
+	podLogString := ""
+	var startTime, endTime time.Time
+	clientset, err := kubernetes.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return "", err
+	}
+
+	if Pod.Status.Phase == v1.PodRunning {
+		req := clientset.CoreV1().Pods(caaPod.ObjectMeta.Namespace).GetLogs(caaPod.ObjectMeta.Name, &v1.PodLogOptions{})
+		podLogs, err := req.Stream(ctx)
+		if err != nil {
+			return "", err
+		}
+		defer podLogs.Close()
+		buf := new(bytes.Buffer)
+		_, err = io.Copy(buf, podLogs)
+		if err != nil {
+			return "", err
+		}
+		podLogString = buf.String()
+
+		if podLogString != "" {
+			podLogSlice := reverseSlice(strings.Split(podLogString, "\n"))
+			for _, i := range podLogSlice {
+				if strings.Contains(i, "calling PullImage for \""+Pod.Spec.Containers[0].Image+"\"") {
+					timeString, err := timeExtractor(i)
+					if err != nil {
+						return "", err
+					}
+					startTime, err = time.Parse("15:04:05", timeString)
+					if err != nil {
+						return "", err
+					}
+					break
+				}
+				if strings.Contains(i, "successfully pulled image \""+Pod.Spec.Containers[0].Image+"\"") {
+					timeString, err := timeExtractor(i)
+					if err != nil {
+						return "", err
+					}
+					endTime, err = time.Parse("15:04:05", timeString)
+					if err != nil {
+						return "", err
+					}
+				}
+			}
+		} else {
+			return "", errors.New("Pod Failed to Start")
```

Comment:
I think we need a different error message string here since in this case the pod is in the Running state but somehow has produced no pod logs.